python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -387,31 +388,55 @@ class RandomForestClassifier(Base):
                              " please read the documentation")
 
     def _obtain_treelite_handle(self):
-        task_category = CLASSIFICATION_MODEL
+        cdef ModelHandle cuml_model_ptr = NULL
+        cdef RandomForestMetaData[float, int] *rf_forest = \
+            <RandomForestMetaData[float, int]*><size_t> self.rf_forest
         if self.num_classes > 2:
             raise NotImplementedError("Pickling for multi-class "
                                       "classification models is currently not "
                                       "implemented. Please check cuml issue "
                                       "#1679 for more information.")
+        if self.treelite_handle is None:
+            task_category = CLASSIFICATION_MODEL
+            build_treelite_forest(
+                & cuml_model_ptr,
+                rf_forest,
+                <int> self.n_cols,
+                <int> task_category,
+                <vector[unsigned char] &> self.model_pbuf_bytes)
+            mod_ptr = <size_t> cuml_model_ptr
```

Comment:
Would use `uintptr_t` instead of `size_t` for pointers cast to integer types. There are a few such occurrences of this in the code. Just pointing out this one.

```suggestion
            mod_ptr = <uintptr_t> cuml_model_ptr
```